### PR TITLE
Support PIL Images in pc.image

### PR DESCRIPTION
**Pull request by Alek99**
### All Submissions:

- [ ] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [ ] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [ ] Does your submission pass the tests? 
- [ ] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.

    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pynecone/utils/format.py [hunk 1]
```diff
@@ -358,6 +361,14 @@ def format_dataframe_values(value: Type) -> List[Any]:
 
     return format_data
 
+def formant_image_data(value: Type) -> str:
```
Comment (TaiJuWu) [inline pynecone/utils/format.py:364]: Is it format_image_data instead of formant_image_data?

**Comment by Alek99** [comment]
```python
"""Welcome to Pynecone! This file create a counter app."""
import pynecone as pc
from PIL import Image
import random
import requests
class State(pc.State):
    """The app state."""


    image = Image.open('assets/favicon.ico')

    def get_random_image(self):
        """Get a random image from unsplash."""
        # Get a random image url from unsplash.
        random_int = random.randint(0, 1000)
        url = f"https://picsum.photos/id/{random_int}/200/300"

        #Update the image.
        self.image = Image.open(requests.get(url, stream=True).raw)
    
def index():
    """The main view."""
    
    return pc.vstack(
        pc.image(src=State.image, alt="random unsplash image"),
        pc.button(
            "Get random image",
            on_click=State.get_random_image,
        ),
    )


# Add state and page to the app. 
app = pc.App(state=State)
app.add_page(index, title="Counter")
app.compile()

```

File: pynecone/components/media/image.py [hunk 2]
```diff
@@ -38,7 +40,7 @@ class Image(ChakraComponent):
     loading: Var[str]
 
     # The image src attribute.
-    src: Var[str]
+    src: Any
```
Comment (picklelo) [inline pynecone/components/media/image.py:43]: I think this can be `Var[Any]`

File: pynecone/components/media/image.py [hunk 3]
```diff
@@ -50,3 +52,11 @@ def get_triggers(self) -> Set[str]:
             The event triggers.
         """
         return super().get_triggers() | {"on_error", "on_load"}
+
+    def _render(self) -> Tag:
+        # If given a pandas df break up the data and columns
```
Comment (picklelo) [inline pynecone/components/media/image.py:57]: Comment needs to be updated

File: pynecone/components/media/image.py [hunk 4]
```diff
@@ -50,3 +52,11 @@ def get_triggers(self) -> Set[str]:
...
+        if types.is_image(type(self.src)):
+            self.src = Var.create(format.formant_image_data(self.src))  # type: ignore
```
Comment (picklelo) [inline pynecone/components/media/image.py:59]: I think we can remove the `type: ignore` after changing to `Var[Any]`

File: pynecone/utils/format.py (hunk 1 above)
Comment (picklelo) [reply pynecone/utils/format.py:364]: +1

File: pynecone/utils/format.py [hunk 5]
```diff
@@ -359,6 +361,23 @@ def format_dataframe_values(value: Type) -> List[Any]:
     return format_data
 
 
+def formant_image_data(value: Type) -> str:
+    """Format image data.
+
+    Args:
+        value: The value to format.
+
+
```
Comment (picklelo) [inline pynecone/utils/format.py:370]: nit: extra blank line

**Review by picklelo — COMMENTED** [review]
Awesome works well for me! Just some minor comments

File: pynecone/components/media/image.py (hunk 4 above)
Comment (Alek99) [reply pynecone/components/media/image.py]: We need it here

**Review by picklelo — APPROVED** [review]
Awesome LGTM. Can you update the PR/commit message before merging

**Pull request merged**